pkg/pillar/cmd/volumemgr/handlecontent.go

```diff
@@ -189,13 +195,60 @@ func createContentTreeStatus(ctx *volumemgrContext, config types.ContentTreeConf
 				}
 				publishBlobStatus(ctx, rootBlob)
 			}
+			AddBlobsToContentTreeStatus(ctx, status, strings.ToLower(config.ContentSha256))
 		}
 	}
 	publishContentTreeStatus(ctx, status)
 	log.Infof("createContentTreeStatus for %v Done", config.ContentID)
 	return status
 }
 
+//AddBlobsToContentTreeStatus adds blob to ContentTreeStatus.Blobs also increments RefCount of the respective BlobStatus.
+//NOTE: This should be the only method to add blobs into ContentTreeStatus.Blobs
+func AddBlobsToContentTreeStatus(ctx *volumemgrContext, status *types.ContentTreeStatus, blobShas ...string) error {
+	log.Infof("AddBlobsToContentTreeStatus(%s): for blobs %v", status.ContentID, blobShas)
+	for _, blobSha := range blobShas {
+		blobStatus := lookupBlobStatus(ctx, blobSha)
+		if blobStatus == nil {
+			err := fmt.Errorf("AddBlobsToContentTreeStatus(%s): No BlobStatus found for blobHash: %s",
+				status.ContentID.String(), blobSha)
+			log.Errorf(err.Error())
+			return err
+		}
+		status.Blobs = append(status.Blobs, blobSha)
+		AddRefToBlobStatus(ctx, blobStatus)
+	}
+	return nil
+}
+
+//RemoveBlobsFromContentTreeStatus removes blob from ContentTreeStatus.Blobs also decrements RefCount of the
+// respective BlobStatus.
+//NOTE: This should be the only method to remove blobs from ContentTreeStatus.Blobs
+func RemoveBlobsFromContentTreeStatus(ctx *volumemgrContext, status *types.ContentTreeStatus, blobShas ...string) error {
+	log.Infof("RemoveBlobsFromContentTreeStatus(%s): for blobs %v", status.ContentID, blobShas)
+	toBeRemovedBlobs := make(map[string]bool)
+	for _, blobSha := range blobShas {
+		toBeRemovedBlobs[blobSha] = true
+	}
+	remainingBlobs := make([]string, 0)
+	for _, blobSha := range status.Blobs {
+		if _, present := toBeRemovedBlobs[blobSha]; present {
+			blobStatus := lookupBlobStatus(ctx, blobSha)
+			if blobStatus == nil {
+				err := fmt.Errorf("RemoveBlobsFromContentTreeStatus(%s): No BlobStatus found for blobHash: %s",
+					status.ContentID.String(), blobSha)
+				log.Errorf(err.Error())
+				return err
```

Comment:
In this error case you need to set Blobs to the ones which have not yet been removed.
But whether or not that is useful depends on the caller(s). What do they do when they get an error? Is the ContentTreeStatus deleted even if this returns an error?

One way would be to log an error and keep going so that status.Blobs is always empty at the end.

Reply:
Changed it to log the error and go ahead to process next blob.